sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AtomicCoder.java

```diff
@@ -35,11 +35,18 @@
   /**
    * {@inheritDoc}.
    *
-   * @throws NonDeterministicException
+   * <p>By default, does not throw.
+   *
+   * @throws NonDeterministicException if the coder is not deterministic
```

Comment:
Suggestion: "@throws NonDeterministicException if overridden to indicate that this subclass of AtomicCoder is not deterministic"

(this should make any weird subclass's javadoc both read well and call out that they should have overridden this javadoc tag)